### DISCUSSION TITLE
make useQuery behave like usePaginatedQuery via config option?

### DISCUSSION THREAD

**Discussion by stefanprobst** [TanStack/query#1168] [Q&A] [answered]
is it possible to somehow configure a `useQuery` hook to behave like `usePaginatedQuery`? i'm thinking of something like `useQuery(key, fn, { paginate: true })`

context: I'm autogenerating hooks from an openapi document, and would love an easy way to change the hook behavior via config - or any other way to get to the previous `resolvedData` while a new request is in-flight. thanks!

**Comment by stefanprobst** [comment, marked as answer]
looks like this will be possible in v3 with #867

**Reply by TkDodo** [reply]
You are correct. See: https://react-query-beta.tanstack.com/guides/migrating-to-react-query-3#usepaginatedquery
